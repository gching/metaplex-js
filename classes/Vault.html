<!doctype html>
<html class="default no-js">
<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>Vault | metaplex-js</title>
	<meta name="description" content="Documentation for metaplex-js">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" href="../assets/css/main.css">
	<script async src="../assets/js/search.js" id="search-script"></script>
</head>
<body>
<header>
	<div class="tsd-page-toolbar">
		<div class="container">
			<div class="table-wrap">
				<div class="table-cell" id="tsd-search" data-index="../assets/js/search.json" data-base="..">
					<div class="field">
						<label for="tsd-search-field" class="tsd-widget search no-caption">Search</label>
						<input id="tsd-search-field" type="text" />
					</div>
					<ul class="results">
						<li class="state loading">Preparing search index...</li>
						<li class="state failure">The search index is not available</li>
					</ul>
					<a href="../index.html" class="title">metaplex-js</a>
				</div>
				<div class="table-cell" id="tsd-widgets">
					<div id="tsd-filter">
						<a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a>
						<div class="tsd-filter-group">
							<div class="tsd-select" id="tsd-filter-visibility">
								<span class="tsd-select-label">All</span>
								<ul class="tsd-select-list">
									<li data-value="public">Public</li>
									<li data-value="protected">Public/Protected</li>
									<li data-value="private" class="selected">All</li>
								</ul>
							</div>
							<input type="checkbox" id="tsd-filter-inherited" checked />
							<label class="tsd-widget" for="tsd-filter-inherited">Inherited</label>
							<input type="checkbox" id="tsd-filter-externals" checked />
							<label class="tsd-widget" for="tsd-filter-externals">Externals</label>
						</div>
					</div>
					<a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a>
				</div>
			</div>
		</div>
	</div>
	<div class="tsd-page-title">
		<div class="container">
			<ul class="tsd-breadcrumb">
				<li>
					<a href="../modules.html">metaplex-js</a>
				</li>
				<li>
					<a href="Vault.html">Vault</a>
				</li>
			</ul>
			<h1>Class Vault</h1>
		</div>
	</div>
</header>
<div class="container container-main">
	<div class="row">
		<div class="col-8 col-content">
			<section class="tsd-panel tsd-hierarchy">
				<h3>Hierarchy</h3>
				<ul class="tsd-hierarchy">
					<li>
						<span class="target">Vault</span>
					</li>
				</ul>
			</section>
			<section class="tsd-panel-group tsd-index-group">
				<h2>Index</h2>
				<section class="tsd-panel tsd-index-panel">
					<div class="tsd-index-content">
						<section class="tsd-index-section ">
							<h3>Constructors</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-constructor tsd-parent-kind-class"><a href="Vault.html#constructor" class="tsd-kind-icon">constructor</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Properties</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Vault.html#accountPublicKey" class="tsd-kind-icon">account<wbr>Public<wbr>Key</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Vault.html#allowFurtherShareCreation" class="tsd-kind-icon">allow<wbr>Further<wbr>Share<wbr>Creation</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Vault.html#authority" class="tsd-kind-icon">authority</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Vault.html#fractionMint" class="tsd-kind-icon">fraction<wbr>Mint</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Vault.html#fractionTreasury" class="tsd-kind-icon">fraction<wbr>Treasury</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Vault.html#key" class="tsd-kind-icon">key</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Vault.html#lockedPricePerShare" class="tsd-kind-icon">locked<wbr>Price<wbr>Per<wbr>Share</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Vault.html#pricingLookupAddress" class="tsd-kind-icon">pricing<wbr>Lookup<wbr>Address</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Vault.html#redeemTreasury" class="tsd-kind-icon">redeem<wbr>Treasury</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Vault.html#state" class="tsd-kind-icon">state</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Vault.html#tokenProgram" class="tsd-kind-icon">token<wbr>Program</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class"><a href="Vault.html#tokenTypeCount" class="tsd-kind-icon">token<wbr>Type<wbr>Count</a></li>
								<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><a href="Vault.html#SCHEMA" class="tsd-kind-icon">SCHEMA</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Accessors</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-get-signature tsd-parent-kind-class"><a href="Vault.html#isAllowFurtherShareCreation" class="tsd-kind-icon">is<wbr>Allow<wbr>Further<wbr>Share<wbr>Creation</a></li>
							</ul>
						</section>
						<section class="tsd-index-section ">
							<h3>Methods</h3>
							<ul class="tsd-index-list">
								<li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><a href="Vault.html#decode" class="tsd-kind-icon">decode</a></li>
							</ul>
						</section>
					</div>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Constructors</h2>
				<section class="tsd-panel tsd-member tsd-kind-constructor tsd-parent-kind-class">
					<a name="constructor" class="tsd-anchor"></a>
					<h3>constructor</h3>
					<ul class="tsd-signatures tsd-kind-constructor tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon">new <wbr>Vault<span class="tsd-signature-symbol">(</span>args<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">Omit</span><span class="tsd-signature-symbol">&lt;</span><a href="Vault.html" class="tsd-signature-type" data-tsd-kind="Class">Vault</a><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">&quot;key&quot;</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">&quot;isAllowFurtherShareCreation&quot;</span><span class="tsd-signature-symbol">&gt;</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="Vault.html" class="tsd-signature-type" data-tsd-kind="Class">Vault</a></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L71">models/programs/token-vault/Vault.ts:71</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>args: <span class="tsd-signature-type">Omit</span><span class="tsd-signature-symbol">&lt;</span><a href="Vault.html" class="tsd-signature-type" data-tsd-kind="Class">Vault</a><span class="tsd-signature-symbol">, </span><span class="tsd-signature-type">&quot;key&quot;</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">&quot;isAllowFurtherShareCreation&quot;</span><span class="tsd-signature-symbol">&gt;</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <a href="Vault.html" class="tsd-signature-type" data-tsd-kind="Class">Vault</a></h4>
						</li>
					</ul>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Properties</h2>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="accountPublicKey" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagOptional">Optional</span> account<wbr>Public<wbr>Key</h3>
					<div class="tsd-signature tsd-kind-icon">account<wbr>Public<wbr>Key<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L38">models/programs/token-vault/Vault.ts:38</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="allowFurtherShareCreation" class="tsd-anchor"></a>
					<h3>allow<wbr>Further<wbr>Share<wbr>Creation</h3>
					<div class="tsd-signature tsd-kind-icon">allow<wbr>Further<wbr>Share<wbr>Creation<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">0</span><span class="tsd-signature-symbol"> | </span><span class="tsd-signature-type">1</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L50">models/programs/token-vault/Vault.ts:50</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="authority" class="tsd-anchor"></a>
					<h3>authority</h3>
					<div class="tsd-signature tsd-kind-icon">authority<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L44">models/programs/token-vault/Vault.ts:44</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="fractionMint" class="tsd-anchor"></a>
					<h3>fraction<wbr>Mint</h3>
					<div class="tsd-signature tsd-kind-icon">fraction<wbr>Mint<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L42">models/programs/token-vault/Vault.ts:42</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="fractionTreasury" class="tsd-anchor"></a>
					<h3>fraction<wbr>Treasury</h3>
					<div class="tsd-signature tsd-kind-icon">fraction<wbr>Treasury<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L46">models/programs/token-vault/Vault.ts:46</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="key" class="tsd-anchor"></a>
					<h3>key</h3>
					<div class="tsd-signature tsd-kind-icon">key<span class="tsd-signature-symbol">:</span> <a href="../enums/VaultKey.html" class="tsd-signature-type" data-tsd-kind="Enumeration">VaultKey</a><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L37">models/programs/token-vault/Vault.ts:37</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="lockedPricePerShare" class="tsd-anchor"></a>
					<h3>locked<wbr>Price<wbr>Per<wbr>Share</h3>
					<div class="tsd-signature tsd-kind-icon">locked<wbr>Price<wbr>Per<wbr>Share<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">BN</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L65">models/programs/token-vault/Vault.ts:65</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="pricingLookupAddress" class="tsd-anchor"></a>
					<h3>pricing<wbr>Lookup<wbr>Address</h3>
					<div class="tsd-signature tsd-kind-icon">pricing<wbr>Lookup<wbr>Address<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L53">models/programs/token-vault/Vault.ts:53</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="redeemTreasury" class="tsd-anchor"></a>
					<h3>redeem<wbr>Treasury</h3>
					<div class="tsd-signature tsd-kind-icon">redeem<wbr>Treasury<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L48">models/programs/token-vault/Vault.ts:48</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="state" class="tsd-anchor"></a>
					<h3>state</h3>
					<div class="tsd-signature tsd-kind-icon">state<span class="tsd-signature-symbol">:</span> <a href="../enums/VaultState.html" class="tsd-signature-type" data-tsd-kind="Enumeration">VaultState</a></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L61">models/programs/token-vault/Vault.ts:61</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="tokenProgram" class="tsd-anchor"></a>
					<h3>token<wbr>Program</h3>
					<div class="tsd-signature tsd-kind-icon">token<wbr>Program<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">PublicKey</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L40">models/programs/token-vault/Vault.ts:40</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class">
					<a name="tokenTypeCount" class="tsd-anchor"></a>
					<h3>token<wbr>Type<wbr>Count</h3>
					<div class="tsd-signature tsd-kind-icon">token<wbr>Type<wbr>Count<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">number</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L60">models/programs/token-vault/Vault.ts:60</a></li>
						</ul>
					</aside>
				</section>
				<section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class tsd-is-static">
					<a name="SCHEMA" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagStatic">Static</span> SCHEMA</h3>
					<div class="tsd-signature tsd-kind-icon">SCHEMA<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">BorshSchema</span><span class="tsd-signature-symbol"> = ...</span></div>
					<aside class="tsd-sources">
						<ul>
							<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L16">models/programs/token-vault/Vault.ts:16</a></li>
						</ul>
					</aside>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Accessors</h2>
				<section class="tsd-panel tsd-member tsd-kind-get-signature tsd-parent-kind-class">
					<a name="isAllowFurtherShareCreation" class="tsd-anchor"></a>
					<h3>is<wbr>Allow<wbr>Further<wbr>Share<wbr>Creation</h3>
					<ul class="tsd-signatures tsd-kind-get-signature tsd-parent-kind-class">
						<li class="tsd-signature tsd-kind-icon"><span class="tsd-signature-symbol">get</span> isAllowFurtherShareCreation<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">boolean</span></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L67">models/programs/token-vault/Vault.ts:67</a></li>
								</ul>
							</aside>
							<h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">boolean</span></h4>
						</li>
					</ul>
				</section>
			</section>
			<section class="tsd-panel-group tsd-member-group ">
				<h2>Methods</h2>
				<section class="tsd-panel tsd-member tsd-kind-method tsd-parent-kind-class tsd-is-static">
					<a name="decode" class="tsd-anchor"></a>
					<h3><span class="tsd-flag ts-flagStatic">Static</span> decode</h3>
					<ul class="tsd-signatures tsd-kind-method tsd-parent-kind-class tsd-is-static">
						<li class="tsd-signature tsd-kind-icon">decode<span class="tsd-signature-symbol">(</span>accountWithPublicKey<span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">AccountWithPublicKey</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="Vault.html" class="tsd-signature-type" data-tsd-kind="Class">Vault</a></li>
					</ul>
					<ul class="tsd-descriptions">
						<li class="tsd-description">
							<aside class="tsd-sources">
								<ul>
									<li>Defined in <a href="https://github.com/gching/metaplex-js/blob/28a99af/src/models/programs/token-vault/Vault.ts#L75">models/programs/token-vault/Vault.ts:75</a></li>
								</ul>
							</aside>
							<h4 class="tsd-parameters-title">Parameters</h4>
							<ul class="tsd-parameters">
								<li>
									<h5>accountWithPublicKey: <span class="tsd-signature-type">AccountWithPublicKey</span></h5>
								</li>
							</ul>
							<h4 class="tsd-returns-title">Returns <a href="Vault.html" class="tsd-signature-type" data-tsd-kind="Class">Vault</a></h4>
						</li>
					</ul>
				</section>
			</section>
		</div>
		<div class="col-4 col-menu menu-sticky-wrap menu-highlight">
			<nav class="tsd-navigation primary">
				<ul>
					<li class=" ">
						<a href="../modules.html">Exports</a>
					</li>
				</ul>
			</nav>
			<nav class="tsd-navigation secondary menu-sticky">
				<ul class="before-current">
				</ul>
				<ul class="current">
					<li class="current tsd-kind-class">
						<a href="Vault.html" class="tsd-kind-icon">Vault</a>
						<ul>
							<li class=" tsd-kind-constructor tsd-parent-kind-class">
								<a href="Vault.html#constructor" class="tsd-kind-icon">constructor</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Vault.html#accountPublicKey" class="tsd-kind-icon">account<wbr>Public<wbr>Key</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Vault.html#allowFurtherShareCreation" class="tsd-kind-icon">allow<wbr>Further<wbr>Share<wbr>Creation</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Vault.html#authority" class="tsd-kind-icon">authority</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Vault.html#fractionMint" class="tsd-kind-icon">fraction<wbr>Mint</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Vault.html#fractionTreasury" class="tsd-kind-icon">fraction<wbr>Treasury</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Vault.html#key" class="tsd-kind-icon">key</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Vault.html#lockedPricePerShare" class="tsd-kind-icon">locked<wbr>Price<wbr>Per<wbr>Share</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Vault.html#pricingLookupAddress" class="tsd-kind-icon">pricing<wbr>Lookup<wbr>Address</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Vault.html#redeemTreasury" class="tsd-kind-icon">redeem<wbr>Treasury</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Vault.html#state" class="tsd-kind-icon">state</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Vault.html#tokenProgram" class="tsd-kind-icon">token<wbr>Program</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class">
								<a href="Vault.html#tokenTypeCount" class="tsd-kind-icon">token<wbr>Type<wbr>Count</a>
							</li>
							<li class=" tsd-kind-property tsd-parent-kind-class tsd-is-static">
								<a href="Vault.html#SCHEMA" class="tsd-kind-icon">SCHEMA</a>
							</li>
							<li class=" tsd-kind-get-signature tsd-parent-kind-class">
								<a href="Vault.html#isAllowFurtherShareCreation" class="tsd-kind-icon">is<wbr>Allow<wbr>Further<wbr>Share<wbr>Creation</a>
							</li>
							<li class=" tsd-kind-method tsd-parent-kind-class tsd-is-static">
								<a href="Vault.html#decode" class="tsd-kind-icon">decode</a>
							</li>
						</ul>
					</li>
				</ul>
				<ul class="after-current">
				</ul>
			</nav>
		</div>
	</div>
</div>
<footer class="with-border-bottom">
	<div class="container">
		<h2>Legend</h2>
		<div class="tsd-legend-group">
			<ul class="tsd-legend">
				<li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li>
				<li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static property</span></li>
				<li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li>
			</ul>
			<ul class="tsd-legend">
				<li class="tsd-kind-property tsd-parent-kind-interface"><span class="tsd-kind-icon">Property</span></li>
			</ul>
		</div>
	</div>
</footer>
<div class="container tsd-generator">
	<p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p>
</div>
<div class="overlay"></div>
<script src="../assets/js/main.js"></script>
</body>
</html>