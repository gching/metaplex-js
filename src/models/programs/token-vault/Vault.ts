import { PublicKey } from '@solana/web3.js';
import BN from 'bn.js';
import { deserializeUnchecked } from 'borsh';
import { BorshSchema } from '../../utils/types';
import { VaultKey } from './VaultKey';
import { AccountWithPublicKey } from '../../../utils/privateTypes';

export enum VaultState {
  Inactive = 0,
  Active = 1,
  Combined = 2,
  Deactivated = 3,
}

export class Vault {
  static SCHEMA: BorshSchema = new Map([
    [
      Vault,
      {
        kind: 'struct',
        fields: [
          ['key', 'u8'],
          ['tokenProgram', 'pubkey'],
          ['fractionMint', 'pubkey'],
          ['authority', 'pubkey'],
          ['fractionTreasury', 'pubkey'],
          ['redeemTreasury', 'pubkey'],
          ['allowFurtherShareCreation', 'u8'],
          ['pricingLookupAddress', 'pubkey'],
          ['tokenTypeCount', 'u8'],
          ['state', 'u8'],
          ['lockedPricePerShare', 'u64'],
        ],
      },
    ],
  ]);
  key = VaultKey.VaultV1;
  accountPublicKey?: PublicKey;
  /// Store token program used
  tokenProgram: PublicKey;
  /// Mint that produces the fractional shares
  fractionMint: PublicKey;
  /// Authority who can make changes to the token-vault
  authority: PublicKey;
  /// treasury where fractional shares are held for redemption by authority
  fractionTreasury: PublicKey;
  /// treasury where monies are held for fractional share holders to redeem(burn) shares once buyout is made
  redeemTreasury: PublicKey;
  /// Can authority mint more shares from fraction_mint after activation
  allowFurtherShareCreation: 0 | 1;

  /// Must point at an ExternalPriceAccount, which gives permission and price for buyout.
  pricingLookupAddress: PublicKey;
  /// In inactive state, we use this to set the order key on Safety Deposit Boxes being added and
  /// then we increment it and save so the next safety deposit box gets the next number.
  /// In the Combined state during token redemption by authority, we use it as a decrementing counter each time
  /// The authority of the token-vault withdrawals a Safety Deposit contents to count down how many
  /// are left to be opened and closed down. Once this hits zero, and the fraction mint has zero shares,
  /// then we can deactivate the token-vault.
  tokenTypeCount: number;
  state: VaultState;

  /// Once combination happens, we copy price per share to token-vault so that if something nefarious happens
  /// to external price account, like price change, we still have the math 'saved' for use in our calcs
  lockedPricePerShare: BN;

  get isAllowFurtherShareCreation(): boolean {
    return this.allowFurtherShareCreation === 1;
  }

  constructor(args: Omit<Vault, 'key' | 'isAllowFurtherShareCreation'>) {
    Object.assign(this, args);
  }

  static decode(accountWithPublicKey: AccountWithPublicKey): Vault {
    const vault = deserializeUnchecked(
      Vault.SCHEMA,
      Vault,
      accountWithPublicKey.account.data,
    ) as Vault;

    return new Vault({
      ...vault,
      accountPublicKey: accountWithPublicKey.pubkey,
    });
  }
}
