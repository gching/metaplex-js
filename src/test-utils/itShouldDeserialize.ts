import { PublicKey } from '@solana/web3.js';
import { serialize } from 'borsh';
import { BorshSchema } from '../models/utils/types';

export function itShouldDeserialize<E>(
  cls: {
    SCHEMA: BorshSchema;
    // eslint-disable-next-line @typescript-eslint/ban-types
    decode: Function;
  },
  received: E,
  accountPubKey = PublicKey.default,
): void {
  const data = Buffer.from(serialize(cls.SCHEMA, received));

  const deserialized = cls.decode({
    pubkey: accountPubKey,
    account: {
      data,
    },
  }) as E;

  expect(deserialized).toDeepEqual(received);
}
