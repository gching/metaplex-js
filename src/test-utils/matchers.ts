import { isEqualWith } from 'lodash';
import BN from 'bn.js';

declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace jest {
    interface Matchers<R> {
      toDeepEqual: <E = any>(expected: E) => R;
    }
  }
}

expect.extend({
  toDeepEqual<E = any>(received: E, expected: E) {
    const pass = isEqualWith(received, expected, (objValue, otherValue) => {
      if (BN.isBN(objValue) && BN.isBN(otherValue)) {
        return objValue.eq(otherValue);
      }
      return undefined;
    });

    if (pass) {
      return {
        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
        message: () => `expected ${received} not to deep equal ${expected}`,
        pass: true,
      };
    }

    return {
      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
      message: () => `expected ${received} to deep equal ${expected}`,
      pass: false,
    };
  },
});

export default undefined;
