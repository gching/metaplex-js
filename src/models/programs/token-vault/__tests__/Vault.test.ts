import { PublicKey } from '@solana/web3.js';
import { itShouldDeserialize } from '../../../../test-utils/itShouldDeserialize';
import { Vault, VaultState } from '../Vault';
import BN from 'bn.js';

describe('Vault', () => {
  it('should instantiate with the correct values', () => {
    const inst = new Vault({
      accountPublicKey: PublicKey.default,
      tokenProgram: PublicKey.default,
      authority: PublicKey.default,
      fractionMint: PublicKey.default,
      fractionTreasury: PublicKey.default,
      pricingLookupAddress: PublicKey.default,
      redeemTreasury: PublicKey.default,
      lockedPricePerShare: new BN(0),
      tokenTypeCount: 0,
      allowFurtherShareCreation: 1,
      state: VaultState.Combined,
    });

    expect(inst).toMatchSnapshot();
  });

  it('should deserialize with the correct values', () => {
    const inst = new Vault({
      accountPublicKey: PublicKey.default,
      tokenProgram: PublicKey.default,
      authority: PublicKey.default,
      fractionMint: PublicKey.default,
      fractionTreasury: PublicKey.default,
      pricingLookupAddress: PublicKey.default,
      redeemTreasury: PublicKey.default,
      lockedPricePerShare: new BN(0),
      tokenTypeCount: 0,
      allowFurtherShareCreation: 1,
      state: VaultState.Combined,
    });

    itShouldDeserialize(Vault, inst);
  });
});
