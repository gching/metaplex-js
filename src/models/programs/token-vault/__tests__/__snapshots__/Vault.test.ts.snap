// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vault should instantiate with the correct values 1`] = `
Vault {
  "accountPublicKey": PublicKey {
    "_bn": "00",
  },
  "allowFurtherShareCreation": 1,
  "authority": PublicKey {
    "_bn": "00",
  },
  "fractionMint": PublicKey {
    "_bn": "00",
  },
  "fractionTreasury": PublicKey {
    "_bn": "00",
  },
  "key": 3,
  "lockedPricePerShare": "00",
  "pricingLookupAddress": PublicKey {
    "_bn": "00",
  },
  "redeemTreasury": PublicKey {
    "_bn": "00",
  },
  "state": 2,
  "tokenProgram": PublicKey {
    "_bn": "00",
  },
  "tokenTypeCount": 0,
}
`;
